hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/key/OMKeyRequest.java

```diff
@@ -351,28 +362,65 @@ protected OmKeyInfo createKeyInfo(@Nonnull KeyArgs keyArgs,
         .setDataSize(size)
         .setReplicationType(type)
         .setReplicationFactor(factor)
-        .setFileEncryptionInfo(encInfo);
+        .setFileEncryptionInfo(encInfo)
+        .setAcls(getAclsForKey(keyArgs, omBucketInfo, prefixManager)).build();
+  }
+
+  private List< OzoneAcl > getAclsForKey(KeyArgs keyArgs,
```

Comment:
Can we move this to some common helper util such as OzoneAclUtil to avoid duplicate code in KeyManagerImpl#getAclsForKey?

Reply:
Discussed offline, leaving this for now.